tracer/src/Datadog.Trace/Iast/Propagation/DefaultInterpolatedStringHandlerModuleImpl.cs

```diff
@@ -91,6 +104,8 @@ public static unsafe void FullTaintIfAnyTainted(IntPtr target, string? input)
 
             var range = new Range(0, result.Length, taintedSelf.Ranges[0].Source, taintedSelf.Ranges[0].SecureMarks);
             taintedObjects.Taint(result, [range]);
+            _taintedRefStructs.Dequeue();
+            taintedSelf.Invalidate();
```

Comment:
I haven't followed this entirely, but are you sure you want a queue and not a stack 🤔

Reply:
Yep, you are right. Changing it